mlflow/server/js/src/gateway/components/endpoints/EndpointsList.tsx

```diff
@@ -0,0 +1,343 @@
+              </div>
+            </TableCell>
+            <TableCell css={{ flex: 1 }}>
+              <ProviderCell modelMappings={endpoint.model_mappings} />
+            </TableCell>
+            <TableCell css={{ flex: 2 }}>
+              <ModelsCell modelMappings={endpoint.model_mappings} />
+            </TableCell>
+            <TableCell css={{ flex: 1 }}>
+              <BindingsCell
+                bindings={getBindingsForEndpoint(endpoint.endpoint_id)}
+                onViewBindings={() =>
+                  setBindingsDrawerEndpoint({
+                    endpointId: endpoint.endpoint_id,
+                    endpointName: endpoint.name ?? endpoint.endpoint_id,
+                    bindings: getBindingsForEndpoint(endpoint.endpoint_id),
+                  })
+                }
+              />
+            </TableCell>
+            <TableCell css={{ flex: 1 }}>
+              <TimeAgo date={timestampToDate(endpoint.last_updated_at)} />
+            </TableCell>
+            <TableCell css={{ flex: 0, minWidth: 96, maxWidth: 96 }}>
+              <div css={{ display: 'flex', gap: theme.spacing.xs }}>
+                <Link to={GatewayRoutes.getEditEndpointRoute(endpoint.endpoint_id)}>
+                  <Button
+                    componentId="mlflow.gateway.endpoints-list.edit-button"
+                    type="primary"
+                    icon={<PencilIcon />}
+                    aria-label={formatMessage({
+                      defaultMessage: 'Edit endpoint',
+                      description: 'Gateway > Endpoints list > Edit endpoint button aria label',
+                    })}
+                  />
+                </Link>
+                <Button
+                  componentId="mlflow.gateway.endpoints-list.delete-button"
+                  type="primary"
+                  icon={<TrashIcon />}
+                  aria-label={formatMessage({
+                    defaultMessage: 'Delete endpoint',
+                    description: 'Gateway > Endpoints list > Delete endpoint button aria label',
+                  })}
+                  onClick={() => handleDeleteClick(endpoint)}
+                />
+              </div>
+            </TableCell>
+          </TableRow>
+        ))}
+      </Table>
+
+      <EndpointBindingsDrawer
+        open={bindingsDrawerEndpoint !== null}
+        endpointName={bindingsDrawerEndpoint?.endpointName ?? ''}
+        bindings={bindingsDrawerEndpoint?.bindings ?? []}
+        onClose={() => setBindingsDrawerEndpoint(null)}
+      />
+
+      <DeleteEndpointModal
+        open={deleteModalEndpoint !== null}
+        endpoint={deleteModalEndpoint}
+        bindings={deleteModalEndpoint ? getBindingsForEndpoint(deleteModalEndpoint.endpoint_id) : []}
+        onClose={() => setDeleteModalEndpoint(null)}
+        onSuccess={handleDeleteSuccess}
+      />
+    </div>
+  );
+};
+
+const ProviderCell = ({ modelMappings }: { modelMappings: Endpoint['model_mappings'] }) => {
```

Comment:
I prefer to have just one component per tsx file. Shall we define them in separate files? cc: @daniellok-db let me know if this is MLflow/Databricks convention.